lib/tbot/cli/start_workload_identity_jwt.go

```diff
@@ -0,0 +1,113 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cli
+
+import (
+	"fmt"
+	"log/slog"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/tbot/config"
+)
+
+// WorkloadIdentityJWTCommand implements `tbot start workload-identity-jwt` and
+// `tbot configure workload-identity-jwt`.
+type WorkloadIdentityJWTCommand struct {
+	*sharedStartArgs
+	*sharedDestinationArgs
+	*genericMutatorHandler
+
+	// NameSelector is the name of the workload identity to use.
+	// --workload-identity-name foo
+	NameSelector string
+	// LabelSelector is the labels of the workload identity to use.
+	// --workload-identity-labels x=y,z=a
+	LabelSelector string
+
+	// Audiences is the list of audiences to include in the JWT.
+	Audiences []string
+}
+
+// NewWorkloadIdentityJWTCommand initializes the command and flags for the
+// `workload-identity-jwt` output and returns a struct that will contain the parse
+// result.
+func NewWorkloadIdentityJWTCommand(parentCmd *kingpin.CmdClause, action MutatorAction, mode CommandMode) *WorkloadIdentityJWTCommand {
+	// TODO(noah): Unhide this command when feature flag removed
+	cmd := parentCmd.Command("workload-identity-jwt", fmt.Sprintf("%s tbot with a SPIFFE-compatible JWT SVID output.", mode)).Hidden()
+
+	c := &WorkloadIdentityJWTCommand{}
+	c.sharedStartArgs = newSharedStartArgs(cmd)
+	c.sharedDestinationArgs = newSharedDestinationArgs(cmd)
+	c.genericMutatorHandler = newGenericMutatorHandler(cmd, c, action)
+
+	cmd.Flag(
+		"name-selector",
+		"The name of the workload identity to issue",
+	).StringVar(&c.NameSelector)
+	cmd.Flag(
+		"label-selector",
+		"A label-based selector for which workload identities to issue. Multiple labels can be provided using ','.",
+	).StringVar(&c.LabelSelector)
+	cmd.Flag(
+		"audience",
+		"Specify the audiences to include in the JWT. At least one audience must be specified.",
+	).Required().StringsVar(&c.Audiences)
+
+	return c
+}
+
+// ApplyConfig applies the parsed flags to the bot configuration.
+func (c *WorkloadIdentityJWTCommand) ApplyConfig(cfg *config.BotConfig, l *slog.Logger) error {
+	if err := c.sharedStartArgs.ApplyConfig(cfg, l); err != nil {
+		return trace.Wrap(err)
+	}
+
+	dest, err := c.BuildDestination()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	svc := &config.WorkloadIdentityJWTService{
+		Destination: dest,
+		Audiences:   c.Audiences,
+	}
+
+	switch {
+	case c.NameSelector != "" && c.LabelSelector != "":
+		return trace.BadParameter("workload-identity-name and workload-identity-labels flags are mutually exclusive")
```

Comment:
nit: I think the arguments were renamed?

Reply:
b42cb119aa